### DISCUSSION TITLE
error Cross Origin ?

### DISCUSSION THREAD

**Discussion by 201603007Akhiles** [dearhive/dearflip-js-flipbook#61] [Q&A] [answered]
why i cannot access file and error cross origin. can you help?

**Comment by deepak-ghimire** [comment, marked as answer]
Cross Origin issue occurs **when the requested PDF file is in different domain** from the page that is request the file.

### Different domains can also include:

1. https and http protocol difference. 
2. www and non-www subdomain difference
3. fully different domains

### What can be done?
In the domain/server where the PDF files exists, you have to add proper configuration for CORS so that the request domain gains permission to load the files.

**For Apache Server, add following lines in .htaccess file**

```
Header set Access-Control-Allow-Origin "*" 
Header set Access-Control-Allow-Headers "Range" 
Header set Access-Control-Expose-Headers: "Accept-Ranges, Content-Encoding, Content-Length, Content-Range"
```
More Info: 
https://enable-cors.org/server.html
https://js.dearflip.com/cors-issues-with-pdf/
